gradio/helpers.py

```diff
@@ -797,71 +798,97 @@ def get_color_gradient(c1, c2, n):
             if isinstance(bars_color, str)
             else get_color_gradient(bars_color[0], bars_color[1], bar_count)
         )
-        plt.bar(
+
+        fig = plt.figure(figsize=(5, 1), dpi=200, frameon=False)
+        fig.subplots_adjust(left=0, bottom=0, right=1, top=1)
+        plt.axis("off")
+        plt.margins(x=0)
+
+        barcollection = plt.bar(
             np.arange(0, bar_count),
             samples * 2,
             bottom=(-1 * samples),
             width=bar_width,
             color=color,
         )
-        plt.axis("off")
-        plt.margins(x=0)
+
+        def animate(_):
```

Comment:
If this increases the runtime about a 10% increase in runtime (going off of your numbers), I think it should be configurable by users! Some may not want to pay that penalty. WDYT?